docs/manual/kernel/Vmm.tex

```diff
@@ -0,0 +1,235 @@
+    \node (vmm3) [rectangle, minimum width=2cm, minimum height=1cm, draw=black, fill=gray!10, right =of vmm2] {VMM 2};
+
+    \node (vmdriver1) [rectangle, minimum width=2.5cm, minimum height=1cm, draw=black, fill=blue!20, below = of vmm2] {};
+    \node (vmdriver2) [rectangle, minimum width=2.5cm, minimum height=1cm, draw=black, fill=blue!20, below right=3mm of vmdriver1.north west] {};
+    \node (vmdriver3) [rectangle, minimum width=2.5cm, minimum height=1cm, draw=black, fill=blue!20, below right=3mm of vmdriver2.north west] {Vm Drivers};
+
+    \node (hat) [rectangle, minimum width=10cm, minimum height=5mm, draw=black, fill=red!10, below= 2mm of vmdriver3] {HAT};
+
+    \draw [<->] (vmm1) -- (vmdriver1);
+    \draw [<->] (vmm2) -- (vmdriver1);
+    \draw [<->] (vmm3) -- (vmdriver1);
+
+    \node (hat0) [minimum height=5mm, right = 1.4cm of hat.west] {};
+    \node (hat1) [minimum height=5mm, right = 5mm of hat0] {};
+    \node (hat2) [minimum height=5mm, left = 2.3cm of hat.east] {};
+    \node (vmm2fudge) [rectangle, minimum width=5mm, minimum height=1cm, left =1mm of vmm2.center] {};
+
+    \draw [->, red] (vmm1) -- (hat0);
+    \path [->, in=90, out=210, red] (vmm2fudge.south) edge (hat1);
+    \draw [->, red] (vmm3) -- (hat2);
+\end{tikzpicture}
+\caption{Relationship between multiple VMMs, VM drivers and the HAT.}
+\end{figure}
+
+\subsection{Concepts}
+
+\paragraph{VM Range}
+Used internally by the virtual memory subsystem to represent a range of virtual addresses/block of virtual memory. It contains the usual base and length fields, as well some flags describing the permissions (user or kernel) and protection (read, write, execute). It also contains an offset field as the virtual address returned may not always point to the start of the VM range. This is to allow things like the VMM adding guard pages to the range, which are included as part of the VM range but not included in the virtual memory made available to the consumer. This also allows for virtual memory to map things in ways that might be misaligned according to the MMU, as the offset field provides a byte-sized offset relative to the granularity the MMU provides. Each range also contains a \verb|token| field, which holds an private pointer for the VM driver that is attached to this range.
+
+\paragraph{VM Driver} 
+\label{vmdrivers}
+While these are called VM \textit{drivers} they actually have nothing to do with the device driver subsystem, and are loaded much earlier than that. It might be more accurate to think of them as plugins to the VMM, with each VM driver providing one type of backing memory that can be attached to a VM range.
+
+\begin{itemize}
+    \item Anon VM Driver: Provides general purpose 'working memory'. This is the most common type of virtual memory and ultimately just allocates physical memory and maps it where appropriate. This driver makes heavy use of the page fault handler to perform operations like demand paging and a primitive form copy-on-write.
```

Comment:
couldn't be better to make the `working memory` italic instead of between single quotes, italic and between quotes? 
(is just a question not an issue, maybe just check de differences and go with whatever you prefer)